### ISSUE TITLE
Enable Grpc gzip compression

### ISSUE THREAD

**Issue by SHUBHAMPARIHAR** [improbable-eng/grpc-web#500]
I am using the latest version of the package i.e. 0.9.6.
I was trying to use Grpc gzip compression. I was able to handle compression on the backend but unable to do the same on the frontend. I am getting this error in the response headers
> grpc-message: grpc: Decompressor is not installed for grpc-encoding "gzip"

Below is my code snippet:
```
let metadata = new grpc.Metadata();
metadata.append('content-encoding', 'gzip')
metadata.append('grpc-encoding', 'gzip');
  grpc.invoke(Auth, {
     request: request,
     metadata: metadata,
     host: '',
     onHeaders: (headers: grpc.Metadata) => {
     },
     onMessage: (message: Auth) => {
     },
     onEnd: (code: grpc.Code, message: string, trailers: grpc.Metadata) => {
     },
  });
```

**Comment by johanbrandhorst** [comment]
Is the error coming from the proxy or the Java server?

**Comment by SHUBHAMPARIHAR** [comment]
I'm not sure about that. Any way I could check this? I'm getting this in the network tab of Chrome

**Comment by johanbrandhorst** [comment]
You could try intercepting the traffic between the proxy and the java server to see what is being returned from the java server. Are the requests reaching the java server at all?

**Comment by SHUBHAMPARIHAR** [comment]
Yes you are correct. I don't think the call is reaching the back-end. I am sending the `grpc-encoding: 'gzip` request headers from the front-end but I think the proxy isn't understanding or reading perfectly. Any way I can handle the proxy to accept the `grpc-encoding` header?

**Comment by johanbrandhorst** [comment]
I don't know if we have support for grpc-encoding right now (or if we need to add it explicitly). The proxy mostly just performs translation from HTTP to gRPC requests, it doesn't try to parse the request or response bodies. Maybe you could try adding some log messages to the proxy?

**Comment by IanDuncanson** [comment]
@johanbrandhorst , we are seeing the same error - ' Decompressor is not installed for grpc-encoding "gzip" '. Our backend grpc service is written in C++. If I register the gzip decompressor in backend.go, then it does seem to get through, but then we are also getting warnings (I think on the incoming https proxy endpoint) - "Compression algorithm ('gzip') not present in the bitset of accepted encodings". I presume this warning is because the browser client is adding this encoding, but the grpcweb proxy isn't declaring that it supports gzip? I am not very well versed in the nuances of encoding and compression so apologies if this doesn't make sense.

**Comment by johanbrandhorst** [comment]
I couldn't say without a clear reproduction example. I haven't run into this issue myself.

**Comment by ajaykMT** [comment]
@IanDuncanson were you able to figure out resolution for "Compression algorithm ('gzip') not present in the bitset of accepted encodings" ?
